### PR TITLE
[AG-364] integrate Lido eth-steth

### PR DESCRIPTION
**Pull request by it4rb**
For Lido ETH->stETH we need to get 2 reserved amounts (totalShares and totalPooledEther)

Formula in their contract code:
```
/**
     * @return the amount of shares that corresponds to `_ethAmount` protocol-controlled Ether.
     */
    function getSharesByPooledEth(uint256 _ethAmount) public view returns (uint256) {
        return _ethAmount
            .mul(_getTotalShares())
            .div(_getTotalPooledEther());
    }
```

### CODE REVIEW

File: pkg/source/lido/pools/ethereum.json [hunk 1]
```diff
@@ -2,6 +2,7 @@
   {
     "id": "0x7f39c581f595b53c5cb19bd0b3f8da6c935e2ca0",
     "name": "wstETH",
+    "type": "lido",
```
Comment (piavgh) [inline pkg/source/lido/pools/ethereum.json:5]: to make it similar to the one below, can we set this as `lido-steth-wsteth`? Is there any problem by doing so?

Comment (it4rb) [reply pkg/source/lido/pools/ethereum.json:5]: um I tried that before but the existed Lido pool didn't get updated so I decided to keep it the same

File: pkg/source/lido-steth/pool_tracker.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	reserves, err := d.getPoolReserves(ctx, p)
+	if err != nil {
+		log.WithFields(logger.Fields{
+			"error": err,
+		}).Error("failed to getPoolReserves")
+		return entity.Pool{}, err
+	}
+
+	p.Reserves = reserves
+	p.Extra = "{}"
```
Comment (hanguyenk) [inline pkg/source/lido-steth/pool_tracker.go:40]: [small] I think we don't need to specify extra in this case

File: pkg/source/lido-steth/pools_list_updater.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+func NewPoolsListUpdater(cfg *Config) *PoolsListUpdater {
+	return &PoolsListUpdater{
+		config: cfg,
+	}
+}
+
+func (d *PoolsListUpdater) GetNewPools(ctx context.Context, metadataBytes []byte) ([]entity.Pool, []byte, error) {
+	if d.hasInitialized {
+		logger.Info("skip since pool has been initialized")
```
Comment (hanguyenk) [inline pkg/source/lido-steth/pools_list_updater.go:28]: [small] I think this log should be in debug level

File: pkg/source/lido-steth/pools_list_updater.go [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+	"time"
+
+	"github.com/KyberNetwork/logger"
+
+	"github.com/KyberNetwork/kyberswap-dex-lib/pkg/entity"
+)
+
+type PoolsListUpdater struct {
+	config         *Config
+	hasInitialized bool
```
Comment (hanguyenk) [inline pkg/source/lido-steth/pools_list_updater.go:17]: If the runner pod is restarted, we will initialize pools all over again? 
Personally I think this behavior is fine as long as it doesn't overwrite the existing ones in Redis.

Comment (it4rb) [reply pkg/source/lido-steth/pools_list_updater.go:17]: yes, we'll reinitialize on restart, and pool-service will skip existed pool without writing to redis https://github.com/KyberNetwork/pool-service/blob/main/internal/runner/pools_list_updater.go#L72

File: pkg/source/lido-steth/pool_tracker.go (hunk 2 above)
Comment (it4rb) [inline pkg/source/lido-steth/pool_tracker.go:40]:
```suggestion
```

File: pkg/source/lido-steth/pools_list_updater.go (hunk 3 above)
Comment (it4rb) [inline pkg/source/lido-steth/pools_list_updater.go:28]:
```suggestion
		logger.Debug("skip since pool has been initialized")
```

**Review by hanguyenk — APPROVED** [review]

**Pull request merged**